heroku/deploy-a-django-app.md

```diff
@@ -0,0 +1,365 @@
+    'version': 1,
+    'disable_existing_loggers': False,  # Preserve default loggers
+    'handlers': {
+        'console': {
+            'class': 'logging.StreamHandler',
+        },
+    },
+    'loggers': {
+        'django': {
+            'handlers': ['console'],
+            'level': os.getenv('DJANGO_LOG_LEVEL', 'INFO'),
+        },
+    },
+}
+```
+
+For more detail on Django's logging framework, [see the documentation](https://docs.djangoproject.com/en/2.2/topics/logging/).
+
+## Provision Heroku resources
+
+Once your application is properly configured for Heroku, the following instructions
+will help you deploy your application to the platform.
+
+### Ensure that the Heroku CLI is installed
+
+The fastest way to get a project up and running on Heroku is to use the
+[Heroku CLI](https://devcenter.heroku.com/articles/heroku-cli). Before you start,
+make sure you have the CLI installed locally. In addition, confirm that you have the
+manifest plugin) installed:
+
+```
+heroku plugins | grep manifest
+```
+
+If you don't see any output, [follow the official instructions for installing the
+plugin](https://devcenter.heroku.com/changelog-items/1441).
+
+### Create Heroku config files
+
+Define config files relative to the root of your repo.
+
+#### `heroku.yml`
+
+The `heroku.yml` manifest file tells Heroku how to build and network the services
+and containers that comprise your application. This file should live in the root of your project repo.
+For background and syntax, see [the
+documentation](https://devcenter.heroku.com/articles/build-docker-images-heroku-yml).
+Use the following baseline to get started:
+
+```yml
+# Define addons that you need for  your project, such as Postgres, Redis, or Solr.
+setup:
+  addons:
+    - plan: heroku-postgresql
+# Define your application's Docker containers.
+build:
+  docker:
+    web: Dockerfile
+# Define any scripts that you'd like to run every time the app deploys.
+release:
+  command:
+    - ./scripts/release.sh
+  image: web
+# The command that runs your application. Replace 'app' with the name of your app.
+run:
+  web: gunicorn -t 180 --log-level debug app.wsgi:application
+```
+
+#### `release.sh`
+
+In your app's `scripts` folder, define a script `release.sh` to run every time the app deploys.
+Use the following baseline script and make sure to run `chmod u+x scripts/release.sh`
+to make it executable:
+
+```bash
+#!/bin/bash
+python manage.py collectstatic --noinput
+python manage.py migrate --noinput
+```
+
+If your app uses a `dbload` script to load initial data into the database, you can use `release.sh`
+to check if the initial data exists and run the data loading scripts if not. For example:
+
+```bash
+# Set ${TABLE} to the name of a table that you expect to have data in it.
+if [ `psql ${DATABASE_URL} -tAX -c "SELECT COUNT(*) FROM ${TABLE}"` -eq "0" ]; then
+    make all
+fi
+```
+
+Note that logs for the release phase won't be viewable in the Heroku console unless `curl`
```

Comment:
What do you think about adding a new heading for Dockerfile edits in the app config section, which would include (1) running collectstatic and (2) installing curl?

Reply:
Done! I added these suggestions to the `Containerize your application` section. I don't love the written description of `RUN` commands, so I'm really looking forward to making some Heroku-focused application templates soon.